### DISCUSSION TITLE
Do video frames have to have dimensions that are multiple of 16 or another number?

### DISCUSSION THREAD

**Discussion by alexeyabel** [gianni-rosato/svt-av1-psy#106] [Q&A] [answered]
AFAIK older codecs need frame dimensions to be multiples of 16 for optimal encoding. Does this restriction exist with SVT-AV1-PSY or can it be any even number or even uneven?

I use ffmpeg's [cropdetect](https://ffmpeg.org/ffmpeg-filters.html#cropdetect) filter to auto crop video files and it uses multiples of 16 by default. I wonder whether I can change this when using SVT-AV1-PSY.

**Comment by silverbacknet** [comment]
That ffmpeg function is based on the lowest common denominator of encoding to old versions of xvid, which either did not pad at all or padded with random/black pixels. It's wildly outdated, all codecs pad with edge pixels these days, so all that mod16 crop gets you is a smaller size and wonky aspect ratio. (Plus HEVC and AV1 don't even use 16x16 as their base anymore.) You shouldn't notice any significant difference in quality or size between 1904x1072 and 1912x1080, whereas in non-padding codec outer fringe pixels would look much worse.

(1080p being non-mod16 is why everything got on board.)

**Reply by alexeyabel** [reply]
Thank you for the explanation, although I am not entirely sure what your recommendation would be. Can I just crop pixel-perfect (mod1) or would it make sense to at least use mod 2?

**Reply by silverbacknet** [reply]
You cannot use odd resolutions in SVT-AV1. mod2 is required and sufficient.

**Comment by gianni-rosato** [comment, marked as answer]
While @silverbacknet is correct that odd resolutions aren't currently supported by SVT-AV1, they _are_ supported in SVT-AV1-PSY. See our README for more information. Limitations encountered in FFmpeg aren't the same as those encountered within the encoder library.
